examples/generative/dcgan_overriding_train_step.py

```diff
@@ -2,22 +2,26 @@
 Title: DCGAN to generate face images
 Author: [fchollet](https://twitter.com/fchollet)
 Date created: 2019/04/29
-Last modified: 2021/01/01
+Last modified: 2023/12/21
 Description: A simple DCGAN trained using `fit()` by overriding `train_step` on CelebA images.
 Accelerator: GPU
 """
 """
 ## Setup
 """
 
+import keras
 import tensorflow as tf
-from tensorflow import keras
-from tensorflow.keras import layers
+
+from keras import layers
+from keras import ops
 import matplotlib.pyplot as plt
 import os
 import gdown
 from zipfile import ZipFile
 
+seed_generator = keras.random.SeedGenerator(42)
```

Comment:
The seed generator must be attached to the layer. So you should do:

```python
class GAN(keras.Model):
    def __init__(self, **kwargs):
        ...
        self.seed_generator = keras.random.SeedGenerator(1337)
```

Reply:
Got it